rdkit/Chem/MolStandardize/normalize.py

```diff
@@ -137,20 +137,24 @@ def normalize(self, mol):
         :return: The normalized fragment.
         :rtype: :rdkit:`Mol <Chem.rdchem.Mol-class.html>`
         """
+
         log.debug('Running Normalizer')
-        # Normalize each fragment separately to get around quirky RunReactants behaviour
-        fragments = []
-        for fragment in Chem.GetMolFrags(mol, asMols=True):
-            fragments.append(self._normalize_fragment(fragment))
-        # Join normalized fragments into a single molecule again
-        outmol = fragments.pop()
-        for fragment in fragments:
-            outmol = Chem.CombineMols(outmol, fragment)
+        # Update: Aggregated normalized fragments into a single molecule according to the previous code.
+        # Step 1: Normalize each fragment separately to get around quirky RunReactants behaviour
+        # Step 2: Join normalized fragments into a single molecule again (previous intention: bottom-up)
+        NORMALIZE_FRAGMENT = self._normalize_fragment
+
+        MolFrags = Chem.GetMolFrags(mol, asMols=True)
+        outmol = NORMALIZE_FRAGMENT(MolFrags[-1])
+        for i in range(len(MolFrags) - 2, -1, -1):
+            outmol = Chem.CombineMols(outmol, NORMALIZE_FRAGMENT(MolFrags[i])) # from bottom to top
```

Comment:
The original code is a lot easier to read. The naming is also inconsistent with the rest of the code. And why reverse the order of combining the fragments. The order shouldn't matter.

Reply:
At the old code, it did exactly as I did. The order I made here is 100% the same as in the original version

Reply:
> The original code is a lot easier to read. The naming is also inconsistent with the rest of the code. And why reverse the order of combining the fragments. The order shouldn't matter.

I have already mentioned that.